src/index.ts

```diff
@@ -124,15 +130,19 @@ export class ResponseValidationError extends Error {
   }
 }
 
-export const serializerCompiler: FastifySerializerCompiler<ZodAny | { properties: ZodAny }> =
-  ({ schema: maybeSchema }) =>
-  (data) => {
-    const schema: Pick<ZodAny, 'safeParse'> = resolveSchema(maybeSchema);
+export const serializerCompiler: FastifySerializerCompiler<ZodAny | { properties: ZodAny }> = ({
+  schema: maybeSchema,
+}) => {
+  const schema = resolveSchema(maybeSchema);
+  const jsonSchema = zodToJsonSchema(schema as ZodAny, zodToFastJsonStringifySchemaOptions);
+  const stringify = build(jsonSchema as FreeformRecord);
```

Comment:
question: hmmm would there be any purpose in passing the json schema to zodToJsonSchema given the stringified result being returned is an unrelated Zod Error shape? Sorry if I don't quite understand what the purpose of `build` is?